packages/fiori-mcp-server/test/integration/provider/provider.ts

```diff
@@ -0,0 +1,369 @@
+import { validate } from './util/validate';
+import { callTool, getTools } from './mcp-server';
+import { existsSync } from 'fs';
+import { join } from 'path';
+
+const MESSAGES_LOG_FOLDER = join(__dirname, '../logs');
+
+/**
+ * Schema for the configuration of the AICoreApiProvider.
+ */
+const AICoreApiProviderConfig = z.object({
+    model: z
+        .enum([
+            'gpt-4o-mini',
+            'gpt-4o',
+            'gpt-4.1',
+            'gpt-4.1-mini',
+            'gpt-4.1-nano',
+            'o1',
+            'o3',
+            'o3-mini',
+            'o4-mini',
+            'gpt-5',
+            'gpt-5-mini',
+            'gpt-5-nano'
+        ])
+        .default('gpt-4o'),
+    label: z.string().optional(),
+    format: z.enum(['json', 'plain']).optional(),
+    timeout: z.number().default(20000) // Default timeout of 20 seconds
+});
+
+type AICoreApiProviderConfig = z.infer<typeof AICoreApiProviderConfig>;
+
+/**
+ * API provider for SAP AI Core using Azure OpenAI models.
+ */
+export default class AICoreApiProvider implements ApiProvider {
+    readonly config: AICoreApiProviderConfig;
+    readonly label: string;
+
+    private client?: Runnable<BaseLanguageModelInput, AIMessageChunk, AzureOpenAiChatCallOptions>;
+    private tools: DynamicStructuredTool[] = [];
+    private runIndex = 0;
+
+    constructor(options: { config: AICoreApiProviderConfig }) {
+        this.config = AICoreApiProviderConfig.parse(options.config);
+
+        process.env.AICORE_SERVICE_KEY = JSON.stringify(getServiceKeyFromEnv());
+
+        this.label = `SAP AI Core (${this.config.model})`;
+    }
+
+    /**
+     * Lazily initializes and returns the Azure OpenAI chat client with bound tool support.
+     *
+     * @returns Azure OpenAI chat client with bound tool support
+     */
+    private async getClient(): Promise<Runnable<BaseLanguageModelInput, AIMessageChunk, AzureOpenAiChatCallOptions>> {
+        if (!this.client) {
+            const client = new AzureOpenAiChatClient({
+                modelName: this.config.model
+            });
+            this.tools = await getTools();
+            this.client = client.bindTools(this.tools);
+            // Cleanup log folder as client initialization happens once
+            await this.cleanupLog();
+        }
+        return this.client;
+    }
+
+    /**
+     * Returns the provider ID string for this API provider instance.
+     *
+     * @returns The provider ID string.
+     */
+    id() {
+        return `SAP AI Core: ${this.config.model}`;
+    }
+
+    /**
+     * Calculate the cost of the request based on the number of tokens used. Cost is in SAP Compute Units (CU).
+     *
+     * @param promptTokens Number of tokens in the prompt.
+     * @param completionTokens Number of tokens in the completion.
+     * @returns Cost in SAP Compute Units (CU).
+     */
+    calculateCost(promptTokens: number, completionTokens: number): number {
```

Comment:
`Function lacks ending return statement and return type does not include 'undefined'.ts(2366)` on my pc.